cmd/nvidia-ctk/cdi/generate/generate.go

```diff
@@ -276,7 +283,15 @@ func (m command) generateSpec(opts *options) (spec.Interface, error) {
 		nvcdi.WithCSVIgnorePatterns(opts.csv.ignorePatterns.Value()),
 		// We set the following to allow for dependency injection:
 		nvcdi.WithNvmlLib(opts.nvmllib),
-	)
+	}
+
+	if len(opts.skipHook.Value()) > 0 {
+		for _, hook := range opts.skipHook.Value() {
+			initOpts = append(initOpts, nvcdi.WithDisabledHook(nvcdi.HookName(hook)))
+		}
+	}
```

Comment:
Did we want to check for validity of the names, or do we want to add that as a follow-up?

Reply:
let's hold this PR until:
- https://github.com/NVIDIA/nvidia-container-toolkit/pull/947
- https://github.com/NVIDIA/nvidia-container-toolkit/pull/927
get in, then I will think on how to add a name check.

Reply:
> let's hold this PR until:

Should this PR not just work automatically with those PRs -- assuming those hooks can be opted out of using the same option?

Reply:
It now allows for disabling all hooks